bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,577 @@
+     *
+     * @param variableName The variable name
+     * @return The value
+     * @throws HeliosException
+     */
+    public void readValue(String variableName) throws HeliosException {
+        final Semaphore lock = transactionLocks.get(comms.getEndpoint());
+        HeliosVariable v = this.vMap.get(variableName);
+        if (v.hasReadAccess()) {
+            try {
+                lock.acquire(); // will block until lock is available
+            } catch (InterruptedException e) {
+                logger.error("{} encountered Exception when trying to read variable {} from the device: {}",
+                        HeliosEasyControlsHandler.class.getSimpleName(), variableName, e.getMessage());
+                return;
+            }
+            // write variable name to register
+            String payload = v.getVariableString();
+            this.comms.submitOneTimeWrite(new ModbusWriteRegisterRequestBlueprint(
+                    HeliosEasyControlsBindingConstants.UNIT_ID, HeliosEasyControlsBindingConstants.START_ADDRESS,
+                    new ModbusRegisterArray(preparePayload(payload)), true,
+                    HeliosEasyControlsBindingConstants.MAX_TRIES), result -> {
+                        this.comms.submitOneTimePoll(
+                                new ModbusReadRequestBlueprint(HeliosEasyControlsBindingConstants.UNIT_ID,
+                                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS,
+                                        HeliosEasyControlsBindingConstants.START_ADDRESS, v.getCount(),
+                                        HeliosEasyControlsBindingConstants.MAX_TRIES),
+                                pollResult -> {
+                                    lock.release();
+                                    Optional<ModbusRegisterArray> registers = (Optional<ModbusRegisterArray>) pollResult
+                                            .getRegisters();
+                                    if (registers.isPresent()) {
+                                        processResponse(v, registers.get());
+                                    }
+                                }, failureCallback -> {
+                                    handleError("{} encountered error reading from device: {}",
+                                            failureCallback.getCause().getMessage(), lock);
+                                });
+                    }, failureCallback -> {
+                        handleError("{} encountered error writing to device: {}",
+                                failureCallback.getCause().getMessage(), lock);
+                    });
+        }
+    }
+
+    /**
+     * Formats the provided date to a string in the device's configured date format
+     *
+     * @param date the date to be formatted
+     * @return a string in the device's configured date format
+     */
+    private String formatDate(String variableName, ZonedDateTime date) {

```

Comment:
The device has separate variables for date and time as well as for the difference to GMT / UTC.
This method only takes care of the date part.
I'm not sure it's actually necessary to consider the time zone here - do you think there is a use case for this?

Reply:
Yes, when writing ZonedDateTime it can be in whatever timezone really. E.g. If it comes from persistence, the time zone might be UTC or might be local, dependending on implementation details.

What should matter is the instant represented by the object.

It would be a different story if we would be dealing with "naive" datetime objects which does not have a concept of timezone.

Two approaches come to mind:
1) write both the timezone and time part. One could convert ZonedDateTime to local time zone first to avoid any funny time zones ending up to the device 
2) convert ZonedDateTime to local time zone (system default), ignore the timezone and write only the time part 

What do you think?

Reply:
3) I guess you could also read the timezone of the device and convert ZonedDateTime to that

Reply:
I think so, I'm now using a single DateTime item and set all 3 paramets (date, time and GMT offset) in one go.
Also, I've added an Action to set the device's system time to the server system time.